src/registrar/utility/csv_export.py

```diff
@@ -161,29 +160,33 @@ def write_csv(
     # Reduce the memory overhead when performing the write operation
     paginator = Paginator(all_domain_infos, 1000)
 
-    if get_domain_managers and len(all_domain_infos) > 0:
-        # We want to get the max amont of domain managers an
-        # account has to set the column header dynamically
-        max_dm_count = max(len(domain_info.domain.permissions.all()) for domain_info in all_domain_infos)
-        update_columns_with_domain_managers(columns, max_dm_count)
-
-    if should_write_header:
-        write_header(writer, columns)
+    # The maximum amount of domain managers an account has
+    # We get the max so we can set the column header accurately
+    max_dm_count = 0
+    total_body_rows = []
 
     for page_num in paginator.page_range:
         rows = []
         page = paginator.page(page_num)
         for domain_info in page.object_list:
+            # Get count of all the domain managers for an account
+            dm_count = domain_info.domain.permissions.count()
+            if dm_count > max_dm_count:
+                max_dm_count = dm_count
+
             try:
-                row = parse_row(columns, domain_info, security_emails_dict, get_domain_managers)
+                row = parse_row(columns, domain_info, max_dm_count, security_emails_dict, get_domain_managers)
                 rows.append(row)
             except ValueError:
                 # This should not happen. If it does, just skip this row.
                 # It indicates that DomainInformation.domain is None.
                 logger.error("csv_export -> Error when parsing row, domain was None")
                 continue
```

Comment:
**comment** don't get why there is a continue here, python continues on excepted errors anyways